### PR TITLE
CLI Setup for Modifying Tasks

### PR DESCRIPTION
**Pull request by amondal2**
[This is downstream of #19 as I wanted @zsusswein to look at both before merging]

This PR implements logic to modify a given task. It recursively iterates through the keys in the task, prompting the user if they want to modify it, and updates the values accordingly. Also contains logic for list and nested dictionary keys. Also did some mild refactoring to move storage functions into utils/

This is the command I used for testing:
```
confmod modify-task -j Rt-estimation-bf57b6e297ad11efb20c00155d63cada-2024-10-31T17-29-56 -t bf57b6e297ad11efb20c00155d63cada_WY_Influenza_1730395796.json
```

### CODE REVIEW

File: cli/console.py [hunk 1]
```diff
@@ -30,7 +33,7 @@ def check_login():
             )
             raise e
 
-        console.print("[italic green] You are logged into Azure. :sparkles:")
+        console.print("[italic green] :sparkles: You are logged into Azure.")
```
Comment (natemcintosh) [inline cli/console.py:36]: As Zack has already said, I'm not sure Azure deserves the sparkles, but I think this new interface deserves the sparkles anyway

File: cli/console.py [hunk 2]
```diff
@@ -101,22 +104,76 @@ def inspect_task(
...
 
+@app.command("modify-task")
+def modify_task(
+    job_id: Annotated[
+        str, typer.Option("--job-id", "-j", help="Job ID to list tasks for")
+    ],
+    task_filename: Annotated[
+        str, typer.Option("--task-filename", "-t", help="Task filename to inspect")
+    ],
+):
```
Comment (natemcintosh) [inline cli/console.py:127]: This is not something that should happen in this PR, but at some point we should allow the user to pass in storage container as an argument as well.

File: utils/epinow2/functions.py [hunk 3]
```diff
@@ -177,6 +177,24 @@ def generate_task_id(
...
+    task_id: str | None = None,
+    timestamp: int | None = None,
+) -> str:
+    """Updates a task_id with new timestamp.
+    Parameters:
+        task_id: task_id to update
+        timestamp: updated timestamp
+    """
+    try:
+        job_id, state, disease, _ = task_id.split("_")
```
Comment (natemcintosh) [inline utils/epinow2/functions.py:190]: Could you add an example of what the `task_id` should look like as a comment?

**Review by natemcintosh — APPROVED** [review]

**Review by natemcintosh — APPROVED** [review]

**Review by zsusswein — APPROVED** [review]
This looks great!

I think this is good to go as-is. I want to generally ask about the update -> overwrite workflow -- do you see any danger here of accidentally overwriting the wrong thing? I'm probably overthinking things here but want to ask.

**Comment by amondal2** [comment]
@zsusswein nothing is being overwritten here (all updates push a new, timestamped task to storage), so mistakes could be "fixed" by just pushing a new task. we should probably implement some workflow to confirm/reject/edit changes before pushing, but that might take some time to figure out how to build correctly

**Pull request merged**